aten/src/ATen/cuda/cub.cuh

```diff
@@ -257,7 +194,7 @@ inline void inclusive_scan(InputIteratorT input, OutputIteratorT output, ScanOpT
       size_cub,
       at::cuda::getCurrentCUDAStream());
   C10_CUDA_KERNEL_LAUNCH_CHECK();
-  using input_t = std::remove_reference_t<decltype(*input)>;
+  using input_t = std::remove_cv_t<std::remove_reference_t<decltype(*input)>>;
```

Comment:
Why do we need `std::remove_cv_t`?

Reply:
The input should be logically `const` so I've used `const scalar_t *` in `inclusive_sum`, but this template uses `input_t` as the type of a temporary buffer which needs to be mutable.

Reply:
I see. Would `typename std::iterator_traits<InputIteratorT>::value_type` be better than `decltype`?